ydb/docs/en/core/reference/ydb-sdk/topic.md

```diff
@@ -1319,12 +1459,126 @@ When reading starts, the client code must transmit the starting consumer offset
 
 - Java
 
-  Setting the starting offset for reading is not supported in the current state of Java SDK.
+  Starting offset for reading in Java can be set only for AsyncReader.
+  In `StartPartitionSessionEvent` callback `StartPartitionSessionSettings` with desired ReadOffset can be passed to `confirm` method.
+  Offset that should be considered as committed can be set with `setCommittedOffset`.
+
+  ```java
+  @Override
+  public void onStartPartitionSession(StartPartitionSessionEvent event) {
+      event.confirm(StartPartitionSessionSettings.newBuilder()
+              .setReadOffset(lastReadOffset)
+              .setCommitOffset(lastCommitOffset)
+              .build());
+  }
+  ```
 
   The `setReadFrom` setting is used for reading only messages with write timestamps no less than the given one.
 
 {% endlist %}
 
+### Reading without a Consumer {#no-consumer}
+Usually reading progress is saved on server within each Consumer. Though such progres can not be saved if a reader is created without a `Consumer`
+
+{% list tabs %}
+
+- Java
+
+  To read without a Consumer, `withoutConsumer()` method should be called explicitly on `ReaderSettings` Builder:
+
+  ```java
+  ReaderSettings settings = ReaderSettings.newBuilder()
+          .withoutConsumer()
+          .addTopic(TopicReadSettings.newBuilder()
+                  .setPath(TOPIC_NAME)
+                  .build())
+          .build();
+  ```
+  In this case reading progress on server will be lost on PartitionSession restart.
+  To avoid reading from the beginning each time, starting offsets should be set each time on `PartitionSession` start:
```

Comment:
```suggestion
  In this case, reading progress on the server will be lost on the `PartitionSession` restart.
  To avoid reading from the beginning each time, starting offsets should be set on each `PartitionSession` start:
```